### PR TITLE
Added Doc annotations for atlas backup

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/atlas/backup/restores_start.go [hunk 1]
```diff
@@ -151,6 +151,10 @@ func RestoresStartBuilder() *cobra.Command {
 		Short:     "Start a restore job for your project and cluster.",
 		Args:      require.ExactValidArgs(1),
 		ValidArgs: []string{automatedRestore, downloadRestore, pointInTimeRestore},
+		Annotations: map[string]string{
+			"args":     "deliveryType",
+			"nameDesc": "Type of restore job to create. Accepted values include: automated, download, pointInTime.",
+		},
```
Comment (andreaangiolillo) [inline internal/cli/atlas/backup/restores_start.go:157]: [q] is this correct? do we want to create an entry to the array for each value? @gssbzn

Comment (gssbzn) [reply internal/cli/atlas/backup/restores_start.go:157]: I delegate this to @mongodb/docs-cloud-team but my preference for these cases where its a small set of known values we we don't annotate but make the long description explain them but up to docs as this is for their pages

**Review by gssbzn — DISMISSED** [review]
LGTM

File: internal/cli/atlas/backup/restores_list.go [hunk 2]
```diff
@@ -60,6 +60,10 @@ func RestoresListBuilder() *cobra.Command {
 		Aliases: []string{"ls"},
 		Short:   "List cloud backup restore jobs for your project and cluster.",
 		Args:    require.ExactArgs(1),
+		Annotations: map[string]string{
+			"args":     "clusterName",
+			"nameDesc": "Name of the Atlas cluster for which you want to retrieve restore jobs.",
```
Comment (melissamahoney-mongodb) [inline internal/cli/atlas/backup/restores_list.go:65]: I think the second key needs to match the value of the first pair? @gssbzn 

```suggestion
			"clusterNameDesc": "Name of the Atlas cluster for which you want to retrieve restore jobs.",
```

File: internal/cli/atlas/backup/restores_start.go [hunk 3]
```diff
@@ -151,6 +151,10 @@ func RestoresStartBuilder() *cobra.Command {
 		Short:     "Start a restore job for your project and cluster.",
 		Args:      require.ExactValidArgs(1),
 		ValidArgs: []string{automatedRestore, downloadRestore, pointInTimeRestore},
+		Annotations: map[string]string{
+			"args":     "deliveryType",
+			"nameDesc": "Type of restore job to create. Accepted values include: automated, download, pointInTime.",
```
Comment (melissamahoney-mongodb) [inline internal/cli/atlas/backup/restores_start.go:156]: Same question here.
```suggestion
			"deliveryTypeDesc": "Type of restore job to create. Accepted values include: automated, download, pointInTime.",
```

File: internal/cli/atlas/backup/restores_start.go (hunk 1 above)
Comment (melissamahoney-mongodb) [reply internal/cli/atlas/backup/restores_start.go:157]: I'm still debating which way I think works better! For now, let's go with this. We will be fleshing out this content in Q3 and can revisit then.

File: internal/cli/atlas/backup/snapshots/create.go [hunk 4]
```diff
@@ -69,6 +69,10 @@ func CreateBuilder() *cobra.Command {
 		Aliases: []string{"take"},
 		Short:   "Create a backup snapshot for your project and cluster.",
 		Args:    require.ExactArgs(1),
+		Annotations: map[string]string{
+			"args":     "clusterName",
+			"nameDesc": "Name of the Atlas cluster whose snapshot you want to restore.",
```
Comment (melissamahoney-mongodb) [inline internal/cli/atlas/backup/snapshots/create.go:74]:
```suggestion
			"clusterNameDesc": "Name of the Atlas cluster whose snapshot you want to restore.",
```

File: internal/cli/atlas/backup/snapshots/delete.go [hunk 5]
```diff
@@ -51,6 +51,10 @@ func DeleteBuilder() *cobra.Command {
 		Aliases: []string{"rm"},
 		Short:   "Delete a backup snapshot.",
 		Args:    require.ExactArgs(1),
+		Annotations: map[string]string{
+			"args":     "snapshotId",
+			"nameDesc": "Unique identifier of the snapshot you want to delete.",
```
Comment (melissamahoney-mongodb) [inline internal/cli/atlas/backup/snapshots/delete.go:56]:
```suggestion
			"snapshotIdDesc": "Unique identifier of the snapshot you want to delete.",
```

File: internal/cli/atlas/backup/snapshots/describe.go [hunk 6]
```diff
@@ -57,6 +57,10 @@ func DescribeBuilder() *cobra.Command {
 		Use:   "describe <ID>",
 		Short: "Get a specific snapshot for your project.",
 		Args:  require.ExactArgs(1),
+		Annotations: map[string]string{
+			"args":     "snapshotId",
+			"nameDesc": "Unique identifier of the snapshot you want to retrieve.",
```
Comment (melissamahoney-mongodb) [inline internal/cli/atlas/backup/snapshots/describe.go:62]:
```suggestion
			"snapshotIdDesc": "Unique identifier of the snapshot you want to retrieve.",
```

File: internal/cli/atlas/backup/restores_list.go (hunk 2 above)
Comment (andreaangiolillo) [reply internal/cli/atlas/backup/restores_list.go:65]: I thought that `nameDesc` has to stay always the same 🤦

**Review by melissamahoney-mongodb — APPROVED** [review]

**Pull request merged**